src/generators/operationInterfaceGenerator.ts

```diff
@@ -0,0 +1,240 @@
+      fileNames.map(fileName => {
+        return {
+          moduleSpecifier: `./${fileName}`
+        } as ExportDeclarationStructure;
+      })
+    );
+  }
+}
+
+/**
+ * This function creates a file for a given Operation Group
+ */
+function generateOperation(
+  operationGroupDetails: OperationGroupDetails,
+  clientDetails: ClientDetails,
+  project: Project
+): void {
+  const name = normalizeName(operationGroupDetails.name, NameType.File);
+  const operationInterfaceGroupFile = project.createSourceFile(
+    `${clientDetails.srcPath}/operationsInterfaces/${name}.ts`,
+    undefined,
+    { overwrite: true }
+  );
+
+  addImports(operationGroupDetails, operationInterfaceGroupFile, clientDetails);
+  addInterface(
+    operationInterfaceGroupFile,
+    operationGroupDetails,
+    clientDetails
+  );
+}
+
+function getReturnType(
+  operation: OperationDetails,
+  importedModels: Set<string>,
+  modelNames: Set<string>
+) {
+  const responseName = getOperationResponseType(
+    operation,
+    importedModels,
+    modelNames
+  );
+
+  return operation.isLRO
+    ? `Promise<LROPoller<${responseName}>>`
+    : `Promise<${responseName}>`;
+}
+
+/**
+ * Adds an Operation group class to the generated file
+ */
+function addInterface(
+  operationGroupFile: SourceFile,
+  operationGroupDetails: OperationGroupDetails,
+  clientDetails: ClientDetails
+) {
+  let importedModels = new Set<string>();
+
+  let allModelsNames = getAllModelsNames(clientDetails);
+
+  const className = normalizeName(
+    operationGroupDetails.name,
+    NameType.OperationGroup,
+    true /** shouldGuard */
+  );
+
+  const operationGroupClass = operationGroupFile.addInterface({
+    name: className,
+    docs: [`Interface representing a ${className}.`],
+    isExported: true
+  });
+
+  writeOperations(
+    operationGroupDetails,
+    operationGroupClass,
+    importedModels,
+    allModelsNames,
+    clientDetails
+  );
+
+  // Use named import from Models
+  if (importedModels.size) {
+    // Add alias to any model that collides with the class name
+    const namedImports = [...importedModels].map(model => {
+      if (model === className) {
+        return `${model} as ${model}Model`;
+      }
+      return model;
+    });
+
+    operationGroupFile.addImportDeclaration({
+      namedImports,
+      moduleSpecifier: "../models"
+    });
+  }
+}
+
+/**
+ * Write operations implementation, extracted from OperationGroupDetails, to the generated file
+ */
+export function writeOperations(
+  operationGroupDetails: OperationGroupDetails,
+  operationGroupClass: InterfaceDeclaration,
+  importedModels: Set<string>,
+  modelNames: Set<string>,
+  clientDetails: ClientDetails
+) {
+  preparePageableOperations(operationGroupDetails, clientDetails);
+  writeAsyncIterators(
+    operationGroupDetails,
+    clientDetails,
+    operationGroupClass,
+    importedModels
+  );
+  operationGroupDetails.operations.forEach(operation => {
+    if (operation.scope != Scope.Private) {
```

Comment:
should we be more strict?
```suggestion
    if (operation.scope !== Scope.Private) {
```